packages/@expo/cli/src/start/server/metro/runServer-fork.ts

```diff
@@ -95,6 +96,23 @@ export const runServer = async (
     end();
   });
 
+  // Extend the close method to ensure all websocket servers are closed, and connections are terminated
+  const originalClose = httpServer.close.bind(httpServer);
+
+  httpServer.close = function closeHttpServer(callback) {
+    originalClose(callback);
+
+    // Close all websocket servers, including possible client connections (see: https://github.com/websockets/ws/issues/2137#issuecomment-1507469375)
+    for (const endpoint of Object.values(websocketEndpoints) as WebSocketServer[]) {
+      endpoint.close();
+      endpoint.clients.forEach((client) => client.terminate());
```

Comment:
This doesn't appear to account for the HMR socket `'/hot': createWebsocketServer({` https://github.com/expo/expo/blob/8ea118e9fd2576792ef6aab9fc49f2e9ed0d0340/packages/%40expo/cli/src/start/server/metro/runServer-fork.ts#L114-L120

Reply:
nevermind